package/lib/src/beamer_delegate.dart

```diff
@@ -731,6 +788,24 @@ class BeamerDelegate extends RouterDelegate<RouteInformation>
     _updateFromLocation(rebuild: false);
   }
 
+  BeamUpdateGuard? _checkAndApplyUpdateGuards(
+    RouteInformation routeInformation,
+    Object? data,
+  ) {
+    // TODO: make guards async => update would be async and all beamXXX functions
+    // TODO: grand parents are not processed => recursive
+    // TODO: locations cannot of have update guards - we would first
+    //       need to have the effective target location for the given routeInformation
```

Comment:
I think we are ready for this. Like I mentioned above, the end goal is to have async `UpdateGuard`s and non-async `BuildGuard`s. Let's first solve the first TODO and then we'll deal with the second. The third one is OK; `BeamLocation`s are alrgith with having just `BuildGuard`s as their role is _building_ the stack of pages, and not routing per se.

Reply:
@slovnicki I updated the PR.  I think addressed all your comments. please let me know if I missed anything or if you found another thing to address.